client/utils/JsonFormatter.cc

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) [2019] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact Novell, Inc.
+ *
+ * To contact Novell about this file by physical or electronic mail, you may
+ * find current contact information at www.novell.com.
+ */
+
+#include <boost/algorithm/string/join.hpp>
+#include <boost/algorithm/string/replace.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+
+#include "client/utils/JsonFormatter.h"
+
+using namespace std;
+
+namespace snapper
+{
+    namespace cli
+    {
+
+	namespace
+	{
+
+	    string indent(u_int indent_level)
+	    {
+		return string(indent_level * 2, ' ');
+	    }
+
+
+	    string escape(const string& value)
+	    {
+		string fixed_value = value;
+
+		boost::algorithm::replace_all(fixed_value, "\\", "\\\\");
+		boost::algorithm::replace_all(fixed_value, "\"", "\\\"");
+		boost::algorithm::replace_all(fixed_value, "\n", "");
```

Comment:
Um, I meant we should *escape* the newlines, not *lose* them.
And while I am fixing that, I have looked up the other [control char escapes](https://tools.ietf.org/html/rfc8259#section-7):

```suggestion
		boost::algorithm::replace_all(fixed_value, "\b", "\\b");
		boost::algorithm::replace_all(fixed_value, "\f", "\\f");
		boost::algorithm::replace_all(fixed_value, "\n", "\\n");
		boost::algorithm::replace_all(fixed_value, "\r", "\\r");
		boost::algorithm::replace_all(fixed_value, "\t", "\\t");
```